hyperspy/_signals/eels.py

```diff
@@ -224,8 +241,11 @@ def align_zero_loss_peak(
         def substract_from_offset(value, signals):
             for signal in signals:
                 signal.axes_manager[-1].offset -= value
-
-        zlpc = self.estimate_zero_loss_peak_centre(mask=mask)
+        
+        if signal_range:
```

Comment:
Actually, given that the following 4 lines are repeated several times in the code it might be worth it to turn them into a function and call the function instead.

```python

def estimate_zero_loss_peak_centre(s, mask, signal_in_range):
    if signal_range:
        zlpc = s.isig[signal_range[0]:signal_range[1]].estimate_zero_loss_peak_centre(mask=mask)
    else:
        zlpc = s.estimate_zero_loss_peak_centre(mask=mask)

   return zlpc

```